scripts/jenkins/ardana/gerrit/build_test_package.py

```diff
@@ -434,50 +290,48 @@ def build_test_packages(change_ids, obs_linked_project, home_project,
     # 'gerrit_project': Package()
     packages = {}
 
-    # Grab each change for the supplied change_ids. As we go through the
-    # changes the change_ids list may expand with dependencies. These are also
-    # processed. If a change has already been processed we skip it to avoid
-    # circular dependencies.
-    try_branches = ['master']
+    # The target branch associated with the first change is used for
+    # all changes
+    branch = None
+
+    # Grab each change for the supplied change_ids
+    changes = []
     for id in change_ids:
-        if id in processed_changes:
-            # Duplicate dependency, skipping..
-            continue
-        c = GerritChange(id, try_branches)
-        # id might not have been in canonical form
-        if c.id in processed_changes:
-            processed_changes.append(id)
+        c = GerritChange(id, branch=branch)
+        branch = branch or c.branch
+        changes.append(c)
+        # Add the dependent changes to the changes list to process
+        changes.extend(c.get_dependencies())
+
+    # As we go through the changes, the changes list may expand with
```

Comment:
(nit) this happens before now rather than as we go through the list

Reply:
that's correct, I fixed the comment